### PR TITLE
Prevent videoencoder from creating any video file on '-n'

### PR DESCRIPTION
**Pull request by okurz**
Verified locally with openQA.

Test: http://lord.arch/tests/6112#downloads, no video is uploaded as NOVIDEO=1
is set
Regression test: http://lord.arch/tests/6113#downloads, video exists

Related progress issue: https://progress.opensuse.org/issues/15896

### CODE REVIEW

File: videoencoder.cpp [hunk 1]
```diff
@@ -173,11 +175,13 @@ int main(int argc, char* argv[])
...
-    ogg_fp = fopen(option_output, "wb");
-    if (!ogg_fp) {
-        fprintf(stderr, "%s: error: %s\n", option_output,
-            "couldn't open output file");
-        return 1;
+    if (output_video) {
+        ogg_fp = fopen(option_output, "wb");
+        if (!ogg_fp) {
+            fprintf(stderr, "%s: error: %s\n", option_output,
+                    "couldn't open output file");
```
Comment (asmorodskyi) [inline videoencoder.cpp]: Can we extend this with $? to get chance understand what is wrong ?

**Comment by Soulofdestiny** [comment]
I'm missing the 'why' :/

File: videoencoder.cpp (hunk 1 above)
Comment (coolo) [inline videoencoder.cpp]: $? is pretty much $? in C :)

File: videoencoder.cpp [hunk 2]
```diff
@@ -72,11 +72,13 @@ static int theora_write_frame(th_ycbcr_buffer ycbcr, int last)
     }
 
     ogg_stream_packetin(&ogg_os, &op);
+    if (!ogg_fp) {
```
Comment (coolo) [inline videoencoder.cpp]: why would this function even called if not?

Comment (okurz) [inline videoencoder.cpp]: updated, not called anymore and the "if" replaced by "assert" at start of function

**Comment by okurz** [comment]
updated:
* theora_write_frame is not called at all anymore
* made benefit even more explicit in commit message

**Pull request merged**